phalcon/mvc/model/binder.zep

```diff
@@ -0,0 +1,197 @@
+
+/*
+ +------------------------------------------------------------------------+
+ | Phalcon Framework                                                      |
+ +------------------------------------------------------------------------+
+ | Copyright (c) 2011-2016 Phalcon Team (http://www.phalconphp.com)       |
+ +------------------------------------------------------------------------+
+ | This source file is subject to the New BSD License that is bundled     |
+ | with this package in the file docs/LICENSE.txt.                        |
+ |                                                                        |
+ | If you did not receive a copy of the license and are unable to         |
+ | obtain it through the world-wide-web, please send an email             |
+ | to license@phalconphp.com so we can send you a copy immediately.       |
+ +------------------------------------------------------------------------+
+ | Authors: Andres Gutierrez <andres@phalconphp.com>                      |
+ |          Eduar Carvajal <eduar@phalconphp.com>                         |
+ |          Wojciech Ślawski <jurigag@gmail.com>                          |
+ +------------------------------------------------------------------------+
+ */
+
+namespace Phalcon\Mvc\Model;
+
+use Phalcon\Mvc\Controller\BindModelInterface;
+use Phalcon\Mvc\Model\Binder\BindableInterface;
+use Phalcon\Cache\BackendInterface;
+
+/**
+ * Phalcon\Mvc\Model\Binding
+ *
+ * This is an class for binding models into params for handler
+ */
+class Binder implements BinderInterface
+{
+	/**
+	 * Array for storing active bound models
+	 *
+	 * @var array
+	 */
+	protected boundModels = [] { get };
+
+	/**
+	 * Cache object used for caching parameters for model binding
+	 */
+	protected cache;
+
+	/**
+	 * Internal cache for caching parameters for model binding during request
+	 */
+	protected internalCache = [];
+
+	/**
+	 * Phalcon\Mvc\Model\Binder constructor
+	 *
+	 * @param \Phalcon\Cache\BackendInterface cache
+	 */
+	public function __construct(<BackendInterface> cache = null)
+	{
+		let this->cache = cache;
+	}
+
+	/**
+	 * Bind models into params in proper handler
+	 */
+	public function bindToHandler(object handler, array params, string cacheKey, var methodName = null) -> array
+	{
+		var paramKey, className, boundModel, paramsCache;
+
+		if handler instanceof \Closure || methodName != null {
+			let this->boundModels = [];
+			let paramsCache = this->getParamsFromCache(cacheKey);
+			if typeof paramsCache == "array" {
+				for paramKey, className in paramsCache {
+					let boundModel = {className}::findFirst(params[paramKey]);
+					let params[paramKey] = boundModel;
+					let this->boundModels[paramKey] = boundModel;
+				}
+
+				return params;
+			}
+
+			return this->getParamsFromReflection(handler, params, cacheKey, methodName);
+		}
+		else {
```

Comment:
else not needed. just throw Exception if handler is not Closure and methodName is null